src/Chapter1.hs

```diff
@@ -641,7 +660,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n = helper n
+  where
+    helper :: Int -> Int
+    helper n = if div n 10 == 0 then n else helper (div n 10)
```

Comment:
Your implementation is correct :+1: 
And it looks fine. Instead of `if ... then ... else` you can use guards, but that's a matter of taste :slightly_smiling_face: 

Here you can assign `div n 10` to a variable and reuse it.

```suggestion
    helper n =
        let nextN = div n 10
        in if nextN == 0 then n else helper nextN
```

I see that you were confused about the `let ... in` syntax, and you have all rights for that! Let-in can be confusing indeed. In general, nobody encourages you to use `let` more, we just want people to learn Haskell syntax basics in here. In some cases `where` works better, in others `let` are good as well. Most of the time it's a matter of taste. I personally prefer guards instead of `if` and `where` instead of `let` :slightly_smiling_face: